### PR TITLE
Add console reporter (writes to the System.out)

### PR DESCRIPTION
**Pull request by mgazanayi**

### CODE REVIEW

**Comment by jcgay** [comment]
Cool @mgazanayi :)
This is a good start but I think we should offer something more human friendly than just `JSON` to the user, WDYT ?

**Comment by mgazanayi** [comment]
Completely ok with the idea. I'll try doing it as soon as possible. Thanks

**Comment by mgazanayi** [comment]
Here is a try using `StringTemplate`, tell me what you think of @jcgay

**Comment by jcgay** [comment]
This is nice!

I haven't read the code so much but we may have a problem with the parameters displayed in the report (they are still correct in the `html` version on your branch):

```
==================================================================================================
Project: maven-color built in 435,0 ms
Run clean on 2019/01/07 08:04:03 with parameters: profileFormatprofile
==================================================================================================
Projects:
--------------------------------------------------------------------------------------------------
maven-color  200,6 ms
Plugin execution: org.apache.maven.plugins:maven-clean-plugin:3.0.0:clean {execution: default-clean} ran in => 130,8 ms

maven-color-coverage-report  12,21 ms
Plugin execution: org.apache.maven.plugins:maven-clean-plugin:3.0.0:clean {execution: default-clean} ran in => 9,417 ms

maven-color-core  5,417 ms
Plugin execution: org.apache.maven.plugins:maven-clean-plugin:3.0.0:clean {execution: default-clean} ran in => 1,677 ms

maven-color-gossip  4,154 ms
Plugin execution: org.apache.maven.plugins:maven-clean-plugin:3.0.0:clean {execution: default-clean} ran in => 1,648 ms

==================================================================================================
Artifact Downloading 0,000 ns
--------------------------------------------------------------------------------------------------
```

The parameter values are not correctly wrote in: `profileFormatprofile` (instead of `Run clean on 2019/01/07 08:04:03 with parameters: {profileFormat=console,html, profile=true}` in the `html`).

**Comment by jcgay** [comment]
I'm wondering if we could use something like https://github.com/vdmeer/asciitable to have a more concise an aligned output ?

The time are not aligned for example, which is harder to read I think:
```
==================================================================================================
Project: maven-color built in 7,002 s
Run test on 2019/01/07 08:10:56 with parameters: profileFormatprofile
==================================================================================================
Projects:
--------------------------------------------------------------------------------------------------
maven-color-core  4,120 s
Plugin execution: org.apache.maven.plugins:maven-surefire-plugin:2.20.1:test {execution: default-test} ran in => 2,220 s
Plugin execution: org.apache.maven.plugins:maven-compiler-plugin:3.7.0:compile {execution: default-compile} ran in => 951,0 ms
Plugin execution: org.apache.maven.plugins:maven-compiler-plugin:3.7.0:testCompile {execution: default-testCompile} ran in => 638,0 ms
Plugin execution: org.apache.maven.plugins:maven-resources-plugin:3.0.2:resources {execution: default-resources} ran in => 194,3 ms
Plugin execution: org.apache.maven.plugins:maven-resources-plugin:3.0.2:testResources {execution: default-testResources} ran in => 13,20 ms
Plugin execution: org.apache.maven.plugins:maven-enforcer-plugin:1.4.1:enforce {execution: enforce-maven} ran in => 2,048 ms
Plugin execution: org.apache.maven.plugins:maven-toolchains-plugin:1.1:toolchain {execution: default} ran in => 2,257 ms

maven-color-gossip  2,139 s
Plugin execution: org.apache.maven.plugins:maven-surefire-plugin:2.20.1:test {execution: default-test} ran in => 794,8 ms
Plugin execution: org.apache.maven.plugins:maven-compiler-plugin:3.7.0:testCompile {execution: default-testCompile} ran in => 607,0 ms
Plugin execution: org.apache.maven.plugins:maven-compiler-plugin:3.7.0:compile {execution: default-compile} ran in => 597,1 ms
Plugin execution: org.apache.maven.plugins:maven-resources-plugin:3.0.2:resources {execution: default-resources} ran in => 4,663 ms
Plugin execution: org.apache.maven.plugins:maven-resources-plugin:3.0.2:testResources {execution: default-testResources} ran in => 2,320 ms
Plugin execution: org.apache.maven.plugins:maven-enforcer-plugin:1.4.1:enforce {execution: enforce-maven} ran in => 1,844 ms
Plugin execution: org.apache.maven.plugins:maven-toolchains-plugin:1.1:toolchain {execution: default} ran in => 1,856 ms

maven-color  546,7 ms
Plugin execution: org.apache.maven.plugins:maven-enforcer-plugin:1.4.1:enforce {execution: enforce-maven} ran in => 370,5 ms
Plugin execution: org.apache.maven.plugins:maven-toolchains-plugin:1.1:toolchain {execution: default} ran in => 79,96 ms

maven-color-coverage-report  7,792 ms
Plugin execution: org.apache.maven.plugins:maven-enforcer-plugin:1.4.1:enforce {execution: enforce-maven} ran in => 1,381 ms
Plugin execution: org.apache.maven.plugins:maven-toolchains-plugin:1.1:toolchain {execution: default} ran in => 1,885 ms
```

**Comment by mgazanayi** [comment]
Very nice lib! I'll try it for sure. Thanks

**Comment by laeubi** [comment]
Any progress on this?

**Comment by mgazanayi** [comment]
Hello,

I forgot this, will be done by tomorrow.

Best

**Comment by mgazanayi** [comment]
Here is a screenshot of the build of maven-profiler it self ^^
Capture d’écran 2021-12-01 à 08 50 08

PS: I added the dependency to show the download part. I want to use it also to set the width of the table dynamically from the terminal data. I'll check it.

**Comment by laeubi** [comment]
@mgazanayi wow this looks really nice!

**Comment by mgazanayi** [comment]
@laeubi : I just pushed a small fix to have a line drawn to separate module from plugins execution.

**Comment by jcgay** [comment]
Wow nice @mgazanayi ! I will have a look :)

**Comment by mgazanayi** [comment]
@jcgay, did you try to build the PR? It's working locally but failing on the CI, due to String comparison. Any idea?

**Comment by jcgay** [comment]
> 

Not yet but it seems to fail because of number formatting ?
The Locale should be set somewhere ?

**Comment by mgazanayi** [comment]
@jcgay : Oh good catch, didn't see the coma instead of the . at all. Let me check.

**Comment by mgazanayi** [comment]
I think it's the default locale of the CI machine. I'll force the formatting despite of the machine config.
BTW, the HTML output is also using the coma separator, I think that in the CI, it would crash also if the content is checked

**Comment by mgazanayi** [comment]
@jcgay : I added coveralls token in my repo so it can run (even if you made the run optional)

**Comment by jcgay** [comment]
> @jcgay : I added coveralls token in my repo so it can run (even if you made the run optional)

Yeah I think my tentative to skip it when a token is not present is not working... 😅

File: pom.xml [hunk 1]
```diff
@@ -52,6 +52,16 @@
...
+        <dependency>
+            <groupId>de.vandermeer</groupId>
+            <artifactId>asciitable-j7</artifactId>
+            <version>1.0.0</version>
+        </dependency>
+        <dependency>
+            <groupId>org.antlr</groupId>
+            <artifactId>stringtemplate</artifactId>
+            <version>4.0.2</version>
+        </dependency>
```
Comment (jcgay) [inline pom.xml:64]: I think this dependency is not used anymore 😇

File: pom.xml [hunk 2]
```diff
@@ -195,7 +205,7 @@
         </plugins>
     </build>
 
-    <profiles>
+    <!--profiles>
```
Comment (jcgay) [inline pom.xml:208]: To be restored :)

File: src/main/java/fr/jcgay/maven/profiler/reporting/json/ToJson.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+    @Override
+    public JsonObject apply(@Nullable Data context) {
+        JsonObject json = new JsonObject();
+        json.add("name", context.getTopProjectName());
+        json.add("time", ms(context.getBuildTime()));
+        json.add("goals", context.getGoals());
+        json.add("date", context.getFormattedDate());
+        json.add("parameters", context.getParameters().toString());
+
```
Comment (jcgay) [inline src/main/java/fr/jcgay/maven/profiler/reporting/json/ToJson.java:25]: It seems `obj.add("profile_name", context.getProfileName());` has disappeared ?

File: src/test/groovy/fr/jcgay/maven/profiler/reporting/console/ConsoleReporterTest.groovy [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+import org.testng.annotations.Test
+
+
+class ConsoleReporterTest {
+
+    private final ByteArrayOutputStream outContent = new ByteArrayOutputStream();
+
+    @BeforeMethod
+    void setUp() {
+        Locale.setDefault(new Locale("en", "EN"));
```
Comment (jcgay) [inline src/test/groovy/fr/jcgay/maven/profiler/reporting/console/ConsoleReporterTest.groovy:22]: Maybe we could store the default locale to restore it in `tearDown` ?

**Review by jcgay — APPROVED** [review]
👍 

Really great, thx @mgazanayi !
I'm making the changes and merging right away :)

**Pull request closed without merging**